pkg/cmd/backuprestore/backupnoconfig.go

```diff
@@ -0,0 +1,88 @@
+package backuprestore
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/klog/v2"
+)
+
+type backupNoConfig struct {
+	snapshotExist bool
+	backupOptions
+}
+
+func NewBackupNoConfigCommand(errOut io.Writer) *cobra.Command {
+	backupNoConf := &backupNoConfig{
+		snapshotExist: false,
+		backupOptions: backupOptions{errOut: errOut},
+	}
+	cmd := &cobra.Command{
+		Use:   "cluster-backup-no-config",
+		Short: "Backs up a snapshot of etcd database and static pod resources without config",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+					}
+					fmt.Fprint(backupNoConf.errOut, err.Error())
+				}
+			}
+
+			must(backupNoConf.Validate)
+			must(backupNoConf.Run)
+		},
+	}
+	backupNoConf.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (b *backupNoConfig) AddFlags(fs *pflag.FlagSet) {
+	b.backupOptions.AddFlags(fs)
+}
+
+func (b *backupNoConfig) Validate() error {
+	return b.Validate()
+}
+
+func (b *backupNoConfig) Run() error {
+	if !b.snapshotExist {
+		if err := backup(&b.backupOptions); err != nil {
+			klog.Errorf("run: backup failed: [%v]", err)
+		}
+		b.snapshotExist = true
+		klog.Infof("config-dir is: %s", b.configDir)
+		return nil
+	}
+
+	if err := b.copySnapshot(); err != nil {
```

Comment:
why do we need to copy the snapshot? can't we just directly put it where it should be?